sdk/include/reactos/debug.h

```diff
@@ -58,6 +60,78 @@ RtlAssert(
     _In_opt_z_ PCHAR Message
 );
 
+#else /* def NTOS_MODE_USER */
+
+typedef _Return_type_success_(return >= 0) LONG NTSTATUS;
+
+#ifndef NO_STRSAFE
+#include <strsafe.h>
+#endif
+
+static inline
+ULONG
+__cdecl
+W32DbgPrint(_In_z_ _Printf_format_string_ PCSTR Format, ...)
+{
+    va_list va;
+    char szBuff[512];
+
+    va_start(va, Format);
+
+#ifdef NO_STRSAFE
+    wvsprintfA(szBuff, Format, va);
```

Comment:
Why not using the standard C runtime function _vsnprintf instead? (This wvsprintfA would enforce a dependence on user32.dll, which is not always present for win32 user-mode apps.)

Reply:
Used `_vsnprintf`.